core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -355,6 +377,43 @@ write_files:
     content: {{$w.GzippedBase64Content}}
   {{- end }}
 {{- end }}
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+  - path: "/opt/bin/ssm/install-ssm-agent.sh"
+    permissions: 0700
+    content: |
+      #!/bin/bash
+      set -eo pipefail
```

Comment:
try `/bin/bash -c 'set -o pipefail; cat /bigtextfile | head -n1 '; echo $?`

in a nutshell, it fails expressions which do not consume whole input, such as `head`, `grep -q` and many others. It is happening because writers which try to write to closed stdout receive SIGPIPE signal, which in 99.999% of cases is unhandled and therefore triggers process shutdown with non 0 exit code. `pipefail` then uses first non-0 exit code in a pipe as a result of whole expression, which leads to surprising errors together with `set -e` which you also set